static/js/common/fxa-login.js

```diff
@@ -5,8 +5,13 @@
         oauthHost: config.oauthHost,
     });
 
-    function currentPath() {
-        return location.pathname + location.search + location.hash;
+    function nextPath() {
+        var to = new Uri(location).getQueryParamValue('to');
```

Comment:
I'm surprised you don't need `(new Uri(location)).getQueryParamValue(...)` since technically `getQueryParamValue()` is a function that can be instantiated.